components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -324,6 +324,10 @@ private String getIDToken(String clientId, String spTenantDomain, JWTClaimsSet j
                     encryptionAlgorithm, encryptionMethod, spTenantDomain,
                     clientId).serialize();
         } else {
+            if (StringUtils.isNotEmpty(oAuthAppDO.getIdTokenSignatureAlgorithm())) {
+                signatureAlgorithm = OAuth2Util.mapSignatureAlgorithmForJWSAlgorithm(
+                        oAuthAppDO.getIdTokenSignatureAlgorithm());
+            }
```

Comment:
Should not alter the class level signatureAlgorithm variable. If the DefaultIDTokenBuilder object is reused erroneous results will be produced.

Also isn't this resolution of the signature alg applicable for all the places? Better to modify the methods to pass in the sig alg from the entry points

Reply:
assigned to a method local variable - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2197/commits/b9eae93266ce28d2891e7f5372b798fb5634fb5c#diff-0441dee1c46a8372bf791b08b2629354ad01d0dd82d9dfbcb36b0472bfb7b752R320